python/ray/autoscaler/autoscaler.py

```diff
@@ -7,7 +7,7 @@
 import hashlib
 import math
 import os
-import queue
+from six.moves import queue
```

Comment:
Probably not related, but I see the following failure

```
testDelayedLaunchWithFailure (__main__.AutoscalingTest) ... StandardAutoscaler: {'head_node': {'TestProp': 1}, 'idle_timeout_minutes': 5, 'file_mounts': {}, 'auth': {'ssh_user': 'ubuntu', 'ssh_private_key': '/dev/null'}, 'max_workers': 10, 'worker_start_ray_commands': ['start_ray_worker'], 'setup_commands': ['cmd1'], 'head_start_ray_commands': ['start_ray_head'], 'head_setup_commands': ['cmd2'], 'worker_nodes': {'TestProp': 2}, 'cluster_name': 'default', 'target_utilization_fraction': 0.8, 'provider': {'region': 'us-east-1', 'type': 'mock', 'availability_zone': 'us-east-1a'}, 'worker_setup_commands': ['cmd3'], 'docker': {'image': 'example', 'container_name': 'mock'}, 'min_workers': 10}
StandardAutoscaler [2018-06-07 19:40:01.098800]: 0/10 target nodes (0 pending)
 - NodeIdleSeconds: Min=-1 Mean=-1 Max=-1
 - NumNodesConnected: 0
 - NumNodesUsed: 0.0
 - ResourceUsage: 
StandardAutoscaler: Launching 5 new nodes
StandardAutoscaler [2018-06-07 19:40:01.099115]: 0/10 target nodes (5 pending)
 - NodeIdleSeconds: Min=-1 Mean=-1 Max=-1
 - NumNodesConnected: 0
 - NumNodesUsed: 0.0
 - ResourceUsage: 
StandardAutoscaler [2018-06-07 19:40:01.204146]: 0/10 target nodes (5 pending)
 - NodeIdleSeconds: Min=-1 Mean=-1 Max=-1
 - NumNodesConnected: 0
 - NumNodesUsed: 0.0
 - ResourceUsage: 
StandardAutoscaler: Launching 3 new nodes
StandardAutoscaler [2018-06-07 19:40:01.204495]: 0/10 target nodes (8 pending)
 - NodeIdleSeconds: Min=-1 Mean=-1 Max=-1
 - NumNodesConnected: 0
 - NumNodesUsed: 0.0
 - ResourceUsage: 
StandardAutoscaler [2018-06-07 19:40:01.209431]: 3/10 target nodes (0 pending)
 - NodeIdleSeconds: Min=-1 Mean=-1 Max=-1
 - NumNodesConnected: 0
 - NumNodesUsed: 0.0
 - ResourceUsage: 
StandardAutoscaler: Launching 5 new nodes
StandardAutoscaler [2018-06-07 19:40:01.209769]: 3/10 target nodes (5 pending)
 - NodeIdleSeconds: Min=-1 Mean=-1 Max=-1
 - NumNodesConnected: 0
 - NumNodesUsed: 0.0
 - ResourceUsage: 
StandardAutoscaler [2018-06-07 19:40:01.215093]: 8/10 target nodes (0 pending)
 - NodeIdleSeconds: Min=-1 Mean=-1 Max=-1
 - NumNodesConnected: 0
 - NumNodesUsed: 0.0
 - ResourceUsage: 
StandardAutoscaler: Launching 2 new nodes
StandardAutoscaler [2018-06-07 19:40:01.215425]: 8/10 target nodes (2 pending)
 - NodeIdleSeconds: Min=-1 Mean=-1 Max=-1
 - NumNodesConnected: 0
 - NumNodesUsed: 0.0
 - ResourceUsage: 
FAIL
```

```
======================================================================
FAIL: testDelayedLaunchWithFailure (__main__.AutoscalingTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/autoscaler_test.py", line 400, in testDelayedLaunchWithFailure
    self.assertEqual(autoscaler.num_launches_pending.value, 0)
AssertionError: 2 != 0

----------------------------------------------------------------------
Ran 26 tests in 1.326s
```

Reply:
@hartikainen perhaps some of the new tests are still flaky?